llvm/include/llvm/CAS/ActionCache.h

```diff
@@ -77,11 +77,11 @@ class ActionCache {
                                          bool Globally = false) const;
 
   /// Asynchronous version of \c get.
-  void getAsync(
-      const CacheKey &ActionKey, bool Globally,
-      unique_function<void(Expected<std::optional<CASID>>)> Callback) const {
+  void getAsync(const CacheKey &ActionKey, bool Globally,
+                unique_function<void(Expected<std::optional<CASID>>)> Callback,
+                std::shared_ptr<Cancellable> *CancelObj) const {
```

Comment:
Is there any specific reason whey cancel object need to be wrapped inside shared_ptr? I would imagine caller needs to own it after requested, so I guess here it is just to avoid caller freeing the object while CAS plugin can still read/write it?

Also nit: for c++ APIs, would be nice to have a default `nullptr` argument for cancel object so caller don't need to specify if they don't want cancel support.

Reply:
> Is there any specific reason whey cancel object need to be wrapped inside shared_ptr?
Good point, `unique_ptr` works as well, I've changed it to `unique_ptr`.

> would be nice to have a default `nullptr` argument
Done 👍